### ISSUE TITLE
[BR]: ModuleNotFoundError: No module named 'fail2ban' (with new information)

### ISSUE THREAD

**Issue by yn-tadpole** [fail2ban/fail2ban#3837]
### Environment:
- Fail2Ban version 1.1.0 : 
- OS, including release name/version : `Linux 6.6.0-28.0.0.34.oe2403.x86_64 #1 SMP Mon May 27 22:22:46 CST 2024 x86_64 x86_64 x86_64 GNU/Linux`
- Build by source code, without modify anything.

### The issue:

Fail2ban build successful but it can't run.
It shows an error like this:
```
File "/usr/local/bin/fail2ban-server", line 34, in <module>
     from fail2ban.client.fail2banserver import exec_command_line
```
The reason was finally found in the file `/etc/systemd/system/fail2ban.service`
```
...
[Service]
Type=simple
Environment="PYTHONNOUSERSITE=1"
ExecStartPre=/bin/mkdir -p /run/fail2ban
ExecStart=/usr/local/bin/fail2ban-server -xf start
...
```
The line `PYTHONNOUSERSITE=1` causes the fail2ban package to fail to load, even though it is already installed normally in `/usr/local/lib/python3.11/site-packages` (run `pip show fail2ban` can verify it)

After changing `PYTHONNOUSERSITE=0`, everything works fine!

However, the packages installed in `/usr/local/lib/python3.11/site-packages` is obviously not user-sites package. 
After setting `PYTHONNOUSERSITE=1`, it is not possible to load the packages in `/usr/local/lib/python3.11/site-packages`, which seems to be a python issue.

**Comment by sebres** [comment]
> Fail2ban build successful but it can't run.

Are I correct assume, you installed it from source? I guess it is some settings of your python or platform and may be probably fixed with some combination of setup parameters like `--user`, `--prefix=`, `--no-site-packages`, etc (and installing under root/sudo).

> After changing `PYTHONNOUSERSITE=0`, everything works fine!

The line `PYTHONNOUSERSITE=1` in our `fail2ban.service` template was a part of fixes 7e3061e7ace0e973378a17de04a2692142e1a6c1.
You (or your maintainers) can surely change that if it is expected for some system or package for some distro.
I'd not like to enable it by default for several reasons.

> After setting PYTHONNOUSERSITE=1, it is not possible to load the packages in /usr/local/lib/python3.11/site-packages, 

Hmm... I never saw `site-packages` in the python path, regardless `PYTHONNOUSERSITE`...
```
# PYTHONNOUSERSITE=1 python3 -c 'import sys; print(sys.path)'
['', '/usr/lib/python311.zip', '/usr/lib/python3.11', '/usr/lib/python3.11/lib-dynload', '/usr/local/lib/python3.11/dist-packages', '/usr/lib/python3/dist-packages']
# PYTHONNOUSERSITE=0 python3 -c 'import sys; print(sys.path)'
['', '/usr/lib/python311.zip', '/usr/lib/python3.11', '/usr/lib/python3.11/lib-dynload', '/usr/local/lib/python3.11/dist-packages', '/usr/lib/python3/dist-packages']
```
Is it something with virtualenv on that system? Or how exactly you build and did install fail2ban?

See also https://stackoverflow.com/q/72004970
However one can surely set `PYTHONPATH` environment variable in systemd unit too.

Anyway it is pretty 3rd party issue, thus I'll close it.

**Comment by sebres** [comment]
Also checked our [latest release for debians](https://github.com/fail2ban/fail2ban/releases/tag/1.1.0), built with the same setup - it is pretty clear in `dist-packages` and not in `site-packages`

```
$ dpkg -c fail2ban_1.1.0-1~upstream1_all.deb | grep '^drwxr.*\./usr/lib/python3/'
drwxr-xr-x root/root         0 2024-04-25 23:58 ./usr/lib/python3/
drwxr-xr-x root/root         0 2024-04-25 23:58 ./usr/lib/python3/dist-packages/
drwxr-xr-x root/root         0 2024-04-25 23:58 ./usr/lib/python3/dist-packages/fail2ban/
drwxr-xr-x root/root         0 2024-04-25 23:58 ./usr/lib/python3/dist-packages/fail2ban/client/
drwxr-xr-x root/root         0 2024-04-25 23:58 ./usr/lib/python3/dist-packages/fail2ban/compat/
drwxr-xr-x root/root         0 2024-04-25 23:58 ./usr/lib/python3/dist-packages/fail2ban/server/
drwxr-xr-x root/root         0 2024-04-25 23:58 ./usr/lib/python3/dist-packages/fail2ban-1.1.0.1.egg-info/
```

Although it'd build with `-install-layout=deb`, so results to `--buildsystem pybuild` etc and may set different base-directories.

However trying to install from source (as described in https://github.com/fail2ban/fail2ban/wiki/How-to-install-or-upgrade-fail2ban-manually#manual-installation-from-source) in a test container, I also see it installing pretty well in `dist-packages`:
```
$ sudo python3 setup.py install --without-tests --dry-run | grep 'creating .*python'
creating /usr/local/lib/python3.11/dist-packages/fail2ban
creating /usr/local/lib/python3.11/dist-packages/fail2ban/server
creating /usr/local/lib/python3.11/dist-packages/fail2ban/compat
creating /usr/local/lib/python3.11/dist-packages/fail2ban/client
```
